win32ss/user/ntuser/object.c

```diff
@@ -871,3 +871,22 @@ PVOID FASTCALL UserAssignmentUnlock(PVOID *ppvObj)
 
     return pvOld;
 }
+
+VOID FASTCALL UserThreadLock1(PVOID pobj, PTL pTL)
+{
+    pTL->next = gptiCurrent->ptl;
+    pTL->pobj = pobj;
+    gptiCurrent->ptl = pTL;
+    if (pobj)
+        UserReferenceObject(pobj);
+}
+
+PVOID FASTCALL UserThreadUnlock1(VOID)
```

Comment:
Would it be a good idea to pass the `PTL` to this function as well, so that you can assert on `ptl == gptiCurrent->ptl` ?

Reply:
It seems a nice idea. I added an argument to `UserThreadUnlock`.